hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -1718,7 +1718,7 @@ private void buildRecursiveOrderedFkSecondPasses(
 		Set<FkSecondPass> dependencies = isADependencyOf.get( currentTable );
 
 		// bottom out
-		if ( dependencies == null || dependencies.size() == 0 ) {
+		if ( dependencies == null || dependencies.isEmpty()  ) {
```

Comment:
This is a good idea. Could you maybe extract all these specific changes into a separate PR for `main` only?